contracts/Registrar.sol

```diff
@@ -109,31 +102,44 @@ contract Registrar {
 
     constructor(
         ENS _ens,
-        ERC20Burnable _rad,
-        address _admin
+        RadicleToken _rad,
+        address _admin,
+        uint _minCommitmentAge
     ) {
         ens = _ens;
         rad = _rad;
         admin = _admin;
+        minCommitmentAge = _minCommitmentAge;
     }
 
     // --- USER FACING METHODS ---
 
     /// Commit to a future name registration
     function commit(bytes32 commitment) external {
         require(commitments.commited(commitment) == 0, "Registrar::commit: already commited");
-        require(rad.balanceOf(msg.sender) >= fee, "Registrar::register: insufficient rad balance");
 
         rad.burnFrom(msg.sender, fee);
         commitments.commit(commitment);
 
         emit CommitmentMade(commitment, block.number);
     }
 
+    /// Commit to a future name and submit permit in the same transaction
+    function commitWithPermit(bytes32 commitment, address owner, address spender, uint256 value,
```

Comment:
Don't we also need to have a signature on the `commitment`?

Reply:
we do for the 0 user actions pattern. For user initiated registrations `msg.sender` still pays, so it's fine

Reply:
or actually this is not required if we are burning from `msg.sender`

Reply:
hmmmm yes it's true. I think there is still a griefing vector in the 1 tx workflow without a signed commitment, although the attacker is going to end up having to register a lot of domains.... :thinking: